utils/generate-command-code.py

```diff
@@ -94,6 +94,58 @@ def check_command_key_specs(command):
     return True
 
 
+def check_implicit_acl_categories(command):
+    """
+    Validate that ACL categories in JSON files follow rules:
+    
+    Implicit ACL category rules (from setImplicitACLCategories in server.c):
+    1. WRITE flag - WRITE category
+    2. READONLY flag (and NOT SCRIPTING category) - READ category
+    3. ADMIN flag - ADMIN and DANGEROUS categories
+    4. PUBSUB flag - PUBSUB category
+    5. FAST flag - FAST category
+    6. BLOCKING flag - BLOCKING category
+    7. If NOT FAST category - SLOW category
+    """
+    command_flags = set(command.desc.get("command_flags", []))
+    acl_categories = set(command.desc.get("acl_categories", []))
+    
+    errors = []
+    
+    if "WRITE" in command_flags and "WRITE" not in acl_categories:
+        errors.append("WRITE flag requires WRITE category")
+    
+    if "READONLY" in command_flags and "SCRIPTING" not in acl_categories:
+        if "READ" not in acl_categories:
+            errors.append("READONLY flag (without SCRIPTING category) requires READ category")
+    
+    if "ADMIN" in command_flags:
+        if "ADMIN" not in acl_categories:
+            errors.append("ADMIN flag requires ADMIN category")
+        if "DANGEROUS" not in acl_categories:
+            errors.append("ADMIN flag requires DANGEROUS category")
```

Comment:
We can check equivalence ADMIN flag == ADMIN category. For example:

```suggestion
    if ("ADMIN" in command_flags) != ("ADMIN" in acl_categories):
        errors.append("ADMIN flag and ADMIN category must match")
    if "ADMIN" in command_flags and "DANGEROUS" not in acl_categories:
        errors.append("ADMIN flag requires DANGEROUS category")
```

Reply:
Currently there are 2 commands that do not satisfy those conditions, lets keep one way check in ADMIN category, WDYT?
```
Processing json files...
Linking container command to subcommands...
Checking all commands...
command: ROLE - ACL category validation errors:
  - ADMIN flag and ADMIN category must match: flags: {'SENTINEL', 'LOADING', 'STALE', 'NOSCRIPT', 'FAST'} categories: {'DANGEROUS', 'FAST', 'ADMIN'}
command: LASTSAVE - ACL category validation errors:
  - ADMIN flag and ADMIN category must match: flags: {'LOADING', 'STALE', 'FAST'} categories: {'DANGEROUS', 'FAST', 'ADMIN'}
Error: There are errors in the commands check, please check the above logs.
```

Reply:
Or should we make them match flag and category? LASTSAVE would be a breaking change of some sort.

Reply:
OK, let's do only refactoring and no breaking change in this PR.